tiledb/api/c_api/rest/test/unit_capi_rest_profile.cc

```diff
@@ -0,0 +1,101 @@
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * Validates the arguments for the RestProfile C API.
+ */
+
+#define CATCH_CONFIG_MAIN
+#include <test/support/tdb_catch.h>
+
+#include "test/support/src/temporary_local_directory.h"
+
+#include "../rest_profile_api_external.h"
+#include "../rest_profile_api_internal.h"
+
+const char* name_ = tiledb::sm::RestProfile::DEFAULT_NAME.c_str();
+tiledb::sm::TemporaryLocalDirectory tempdir_("unit_capi_rest_profile");
+
+/**
+ * No TEST_CASE "C API: tiledb_rest_profile_alloc argument validation" here;
+ * The user-facing API is not to be used in-test, to avoid async changes to
+ * the local, on-disk file. Instead, `tiledb_rest_profile_alloc_test` is used
+ * throughout this test, including in ordinary_rest_profile.
+ */
+
+TEST_CASE(
+    "C API: tiledb_rest_profile_alloc_test argument validation",
+    "[capi][rest_profile]") {
+  capi_return_t rc;
+  tiledb_rest_profile_t* profile;
+  auto homedir = tempdir_.path().c_str();
+  SECTION("success") {
+    rc = tiledb_rest_profile_alloc_test(name_, homedir, &profile);
+    REQUIRE(tiledb_status(rc) == TILEDB_OK);
+    REQUIRE_NOTHROW(tiledb_rest_profile_free(&profile));
+    CHECK(profile == nullptr);
+  }
+  SECTION("empty name") {
+    rc = tiledb_rest_profile_alloc_test("", homedir, &profile);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null name") {
+    rc = tiledb_rest_profile_alloc_test(nullptr, homedir, &profile);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("empty homedir") {
+    rc = tiledb_rest_profile_alloc_test(name_, "", &profile);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null homedir") {
+    rc = tiledb_rest_profile_alloc_test(name_, nullptr, &profile);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+  SECTION("null profile") {
+    rc = tiledb_rest_profile_alloc_test(name_, homedir, nullptr);
+    REQUIRE(tiledb_status(rc) == TILEDB_ERR);
+  }
+}
+
+TEST_CASE(
+    "C API: tiledb_rest_profile_free argument validation",
+    "[capi][rest_profile]") {
+  tiledb_rest_profile_t* profile;
+  auto rc =
+      tiledb_rest_profile_alloc_test(name_, tempdir_.path().c_str(), &profile);
+  REQUIRE(tiledb_status(rc) == TILEDB_OK);
+  SECTION("success") {
+    REQUIRE_NOTHROW(tiledb_rest_profile_free(&profile));
+    CHECK(profile == nullptr);
+  }
+  SECTION("null profile") {
+    /*
+     * `tiledb_rest_profile_free` is a void function, otherwise we would check
+     * for an error.
+     */
+    REQUIRE_NOTHROW(tiledb_rest_profile_free(nullptr));
```

Comment:
It's a little odd that this doesn't throw.
`ensure_output_pointer_is_valid` checks for NULL I assume?
I suppose it's that we don't throw out of C APIs and the exception doesn't get converted to `Status`.